### PR TITLE
fix(iot-service): Send if-match header in the request to updateTwin

### PR DESCRIPTION
**Pull request by barustum**
This pull request is a fix for [this]( https://github.com/Azure/azure-iot-sdk-java/issues/574) issue.

This change ensures an if-match header is sent along with the request to update the twin. This header ensures that the etag a user sets explicitly matches the one that is set in the twin. If both etags are different, the service should throw a IotHubPreconditionFailedException exception. Without the if-match header, a user can make updates to the device twin without knowing whether another update has been performed from a different client.

### CODE REVIEW

**Comment by abhipsaMisra** [comment]
Note regarding naming commits: https://github.com/Azure/azure-iot-sdk-java/blob/master/.github/CODING_GUIDELINES.md#Commit-Guidelines

**Comment by timtay-microsoft** [comment]
You will need to add an Android runner class to mirror the JVMRunner class you made and a new android test group to run it. You will also need to edit our windowsLinuxAndAndroidBuildMatrixConfig.yaml file to use the new android test group. We can chat about this step if you need help.

You can edit the yaml file to just add a new line for testGroup40 like we have for [testGroup39](https://github.com/Azure/azure-iot-sdk-java/blob/8f1fd3c34cc4937273564ef6160c80b6894c306c/vsts/windowsLinuxAndAndroidBuildMatrixConfig.yaml#L258)

File: iot-e2e-tests/common/src/main/java/com/microsoft/azure/sdk/iot/common/tests/iothub/twin/UpdateTwinTests.java [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+import java.io.IOException;
+import java.net.URISyntaxException;
+import java.security.GeneralSecurityException;
+import java.util.*;
+
+/**
+ * Test class containing all tests to be run on JVM and android pertaining to Queries. Class needs to be extended
+ * in order to run these tests as that extended class handles setting connection strings and certificate generation
+ */
+public class UpdateTwinTests extends DeviceTwinCommon
```
Comment (timtay-microsoft) [inline iot-e2e-tests/common/src/main/java/com/microsoft/azure/sdk/iot/common/tests/iothub/twin/UpdateTwinTests.java]: Extending DeviceTwinCommon will make your tests in this file run with parameterized inputs on device client protocols, which doesn't seem relevant to a service client only test. Could you remove this extension and add in the parts that you need from it manually instead?

Comment (barustum) [reply iot-e2e-tests/common/src/main/java/com/microsoft/azure/sdk/iot/common/tests/iothub/twin/UpdateTwinTests.java]: sure

**Comment by barustum** [comment]
@abhipsaMisra . Thanks for letting me know about the commit naming guidelines. I will fix the name when I squash and merge.

**Comment by barustum** [comment]
Synced up with a couple of team members. We decided to hold off on this fix as it might introduce some complexity for our developers, not to mention it might introduce breaking changes. We gave this issue a lower priority, hence we will address this fix as an enhancement once it's at a higher priority.
